### DISCUSSION TITLE
Using Supabase's cron job but have issues in auth policies!

### DISCUSSION THREAD

**Discussion by Kgitman** [supabase/supabase#28143] [Questions]
Hi @GaryAustin1 I'm using the supabase's `cron` and it's work fine but when I enable the `RLS` on the table than there is error in cron job that `data is not inserted because it's violate the RLS policy`, the by default policy is SELECT policy with access to anon .

i want to continue with RLS but some how that the for cron job access the UPDATE or INSERT but for PUBLIC it's only SELECT.

so how can i do this can you specify?

and for upsert what i have to use upsert or insert?

**Comment by GaryAustin1** [comment]
Not sure why your cron job is getting blocked by RLS.  What user are using to create the cron job?
Normally to avoid RLS in system type tasks initiated by an RLS user you would need to use a security definer function to bypass RLS.

**Reply by GaryAustin1** [reply]
You normally don't create new answers when responding to an existing potential answer, to keep things threaded.

You will need to show some code about what you are doing.  How are you creating a cron task from Deno?  You could also use the Supabase Discord as that is easier for back and forth if needed.
